### PR TITLE
#1015 - Add split method for zonotopes

### LINKED ISSUE #1015: Method to split a zonotope
A method to split a zonotope into two "smaller" zonotopes and its proof is presented in [Prop. 3, 1].

[1] *Althoff, M., Stursberg, O., & Buss, M. (2008). Reachability analysis of
nonlinear systems with uncertain parameters using conservative linearization.
In Proc. of the 47th IEEE Conference on Decision and Control.*

### PR DESCRIPTION
**Pull request by mforets**
Closes #1015.

### CODE REVIEW

File: test/unit_Zonotope.jl [hunk 1]
```diff
@@ -96,4 +96,9 @@ for N in [Float64, Rational{Int}, Float32]
     Z = convert(Zonotope, Hyperrectangle(N[2, 3], N[4, 5]))
     @test Z.center == N[2, 3] && diag(Z.generators) == N[4, 5]
     convert(Zonotope, BallInf(N[5, 3], N(2)))
+
+    # split a zonotope
+    Z = Zonotope(N[0, 0], N[1 1; -1 1])
+    Z1, Z2 = split(Z, 1) # in this case the splitting is exact
+    Z1 ⊆ Z && Z2 ⊆ Z
```
Comment (schillic) [inline test/unit_Zonotope.jl]:
```suggestion
    @test Z1 ⊆ Z && Z2 ⊆ Z
```

File: src/Zonotope.jl [hunk 2]
```diff
@@ -438,6 +439,58 @@ function reduce_order(Z::Zonotope{N}, r)::Zonotope{N} where {N<:Real}
...
+```math
+Z^* = ⟨c, (g^{(1,…,j-1)}, g^{(j+1,…, p)})⟩.
+```
+
+[1] *Althoff, M., Stursberg, O., & Buss, M. (2008). Reachability analysis of
+nonlinear systems with uncertain parameters using conservative linearization.
+In Proc. of the 47th IEEE Conference on Decision and Control.*
+"""
+function split(Z::Zonotope, j::Int)
+    @assert 1 <= j <= ngens(Z) "cannot split along index $j a zonotope with $(ngens(Z)) generators"
```
Comment (schillic) [inline src/Zonotope.jl]:
```suggestion
    @assert 1 <= j <= ngens(Z) "cannot split a zonotope with $(ngens(Z)) generators along index $j"
```

File: src/Zonotope.jl [hunk 3]
```diff
@@ -438,6 +439,58 @@ function reduce_order(Z::Zonotope{N}, r)::Zonotope{N} where {N<:Real}
...
+    c, G = Z.center, Z.generators
+    Gj = G[:, j]
+    Gj_half = Gj / 2
+
+    c₁ = c - Gj_half
+    c₂ = c + Gj_half
+
+    G₁ = copy(G)
+    G₁[:, j] = Gj_half
+    G₂ = G₁
```
Comment (schillic) [inline src/Zonotope.jl]:
```suggestion
    G₂ = copy(G₁)
```

Comment (mforets) [reply src/Zonotope.jl]: oops, this was a real problem 💯

**Pull request merged**